prow/tide/status.go

```diff
@@ -317,7 +330,66 @@ func (sc *statusController) setStatuses(all []PullRequest, pool map[string]PullR
 	}
 }
 
+func (sc *statusController) load() {
+	if sc.path == "" {
+		sc.logger.Debug("No stored state configured")
+		return
+	}
+	entry := sc.logger.WithField("path", sc.path)
+	reader, err := sc.opener.Reader(context.Background(), sc.path)
+	if err != nil {
+		entry.WithError(err).Warn("Cannot open stored state")
+		return
+	}
+	defer io.LogClose(reader)
+
+	buf, err := ioutil.ReadAll(reader)
+	if err != nil {
+		entry.WithError(err).Warn("Cannot read stored state")
+		return
+	}
+
+	var stored storedState
+	if err := yaml.Unmarshal(buf, &stored); err != nil {
+		entry.WithError(err).Warn("Cannot unmarshal stored state")
+		return
+	}
+	sc.storedState = stored
+}
+
+func (sc *statusController) save(ticker *time.Ticker) {
+	for range ticker.C {
+		if sc.path == "" {
+			return
+		}
+		entry := sc.logger.WithField("path", sc.path)
+		current := sc.storedState
+		buf, err := yaml.Marshal(current)
+		if err != nil {
+			entry.WithError(err).Warn("Cannot marshal state")
+			continue
+		}
+		writer, err := sc.opener.Writer(context.Background(), sc.path)
+		if err != nil {
+			entry.WithError(err).Warn("Cannot open state writer")
+			continue
+		}
+		if _, err = writer.Write(buf); err != nil {
+			defer io.LogClose(writer)
+			entry.WithError(err).Warn("Cannot write state")
+			continue
+		}
+		if err := writer.Close(); err != nil {
+			entry.WithError(err).Warn("Failed to close written state")
+		}
```

Comment:
It might be nice to have some debug statements for successful loads and saves.

Reply:
Done